src/preprocessing/passes/bool_to_bv.cpp

```diff
@@ -39,165 +39,159 @@ PreprocessingPassResult BoolToBV::applyInternal(
       options::preprocessStep());
 
   unsigned size = assertionsToPreprocess->size();
-  for (unsigned i = 0; i < size; ++i)
+
+  if (options::boolToBitvector() == BOOL_TO_BV_ALL)
+  {
+    for (unsigned i = 0; i < size; ++i)
+    {
+      Node newAssertion = lowerAssertion((*assertionsToPreprocess)[i], true);
+      assertionsToPreprocess->replace(i, Rewriter::rewrite(newAssertion));
+    }
+  }
+  else if (options::boolToBitvector() == BOOL_TO_BV_ITE)
+  {
+    for (unsigned i = 0; i < size; ++i)
+    {
+      assertionsToPreprocess->replace(
+          i, Rewriter::rewrite(lowerIte((*assertionsToPreprocess)[i])));
+    }
+  }
+  else
   {
-    assertionsToPreprocess->replace(
-        i, Rewriter::rewrite(lowerAssertion((*assertionsToPreprocess)[i])));
+    // Unhandled option
```

Comment:
Since we only have two cases here, I'd suggest to assert that the second case is the case you expect. Avoid using `Unreachable()` when possible.

Reply:
Done with 2b4d2bafd6aa2381b092701c0af9953a1602c428.